envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -18,7 +18,15 @@ option (gogoproto.equal_all) = true;
 
 // Upstream host identifier.
 message Endpoint {
```

Comment:
Do we really need each endpoint to have its own alternative health check port, or can this just be specified at cluster level?

Reply:
Some related discussions:
- https://github.com/envoyproxy/envoy/issues/439
- https://github.com/envoyproxy/envoy/pull/560

From above threads, I perceived that providing an alternative health check port per endpoint is preferred.

Reply:
I think per-host is likely required due to potentially dynamic ports on the service side. In the cluster general situation it's more verbose, but I don't think we are optimizing for lack of verbosity here so I think it's OK?

Reply:
SGTM